src/main/perl/Path.pm

```diff
@@ -544,6 +559,99 @@ sub directory
 }
 
 
+=item _make_link
+
+This method is mainly a wrapper over C<LC::Check::link>
+returning the standard CAF::Path return values. Every option
+supported by C<LC::Check::link> is supported. C<NoActionFlag>
+is handled by C<LC::Check::link>.
+
+This in internal method, not supposed to be called directly.
+Either call C<symlink> or C<hardlink> public methods instead.
+
+=cut
+
+sub _make_link
+{
+    my ($self, $path, $target, %opts) = @_;
+    $path = $self->_untaint_path($path, "symlink") || return;
+
+    $self->_reset_exception_fail('symlink');
+
+    my $status = LC::Check::link($path, $target, %opts);
+
+    if ( defined($status) ) {
+        return ($status ? CHANGED : SUCCESS);
+    } else {
+        my $link_type = $opts{hard} ? "hardlink" : "symlink";
+        $self->fail("Failed to create $link_type $path (target=$target)");
+        return;
+    }
+}
+
+=item hardlink
+
+Create a hardlink C<path> whose target is C<target>.
+
+Returns an error if C<path> already exists and is not a 
+symlink. If it exists and is a symlink, it is updated.
+The target must exist.
+
+Returns SUCCESS on sucess if the symlink already existed
+with the same target, CHANGED if the symlink was created
+or updated, undef otherwise.
+
+This method relies on C<_make_link> method to do the real work,
+after enforcing the option saying that it is a symlink.
+
+=cut
+
+sub hardlink
+{
+    my ($self, $path, $target, %opts) = @_;
+
+    $opts{hard} = 1;
+    my $status = $self->_make_link($path, $target, %opts);
+
+    return $status;
+}
+
+
+=item symlink
+
+Create a symlink C<path> whose target is C<target>.
+
+Returns an error if C<path> already exists and is not a 
+symlink, except if this is a file and option C<force> is
+defined and true. If C<path> exists and is a symlink, it is updated.
+By default, the target is not required to exist. If you want to
```

Comment:
please use `check`, it's not because we use LC internally (for now), we will do so forever or that we should use the same option names or argument order. now is the time to try to make a clean API